youtube_dl/extractor/younow.py

```diff
@@ -0,0 +1,94 @@
+    _TESTS = [
+        {
+            'url': 'https://www.younow.com/MsMastaFoxx_CosmicFam',
+            'info_dict': {
+                'id': 'N608ngRRa1w',
+                'ext': 'mp4',
+                'title': 'I\'m LIVE on YouNow (link in description)',
+                'description': 'CLICK TO WATCH & CHAT LIVE WITH ME:\nhttps://www.younow.com/MsMastaFoxx_CosmicFam/156809157/34576709/c75982c1/b',
+                'uploader': 'Masta Foxx',
+                'uploader_id': 'mindfitprogram',
+                'upload_date': '20170312',
+            },
+            'add_ie': ['Youtube'],
+        },
+        {
+            'url': 'https://www.younow.com/nowthisisliving',
+            'info_dict': {
+                'id': 'nowthisisliving',
+                'ext': 'mp4',
+                'is_live': True,
+                'title': 'March 14, 2017',
+                'description': 'YouNow is the best way to broadcast live and get an audience to watch you.',
+                'thumbnail': 'https://ynassets.s3.amazonaws.com/broadcast/live/156995548/156995548.jpg',
+                'tags': ['girls'],
+                'categories': ['girls'],
+                'uploader': 'nowthisisliving',
+                'uploader_id': '4497161',
+                'uploader_url': 'https://www.younow.com/nowthisisliving',
+                'creator': 'nowthisisliving',
+                'formats': [{
+                    'url': 'https://cdn.younow.com/php/api/broadcast/videoPath/hls=1/broadcastId=156995548/channelId=4497161',
+                    'no_resume': True,
+                    'ext': 'mp4',
+                    'protocol': 'm3u8',
+                }],
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        username = self._match_id(url)
+        data = self._download_json('https://api.younow.com/php/api/broadcast/info/curId=0/user=%s' % (username), username)
+
+        # YouTube-based stream
+        if 'ytBroadcastId' in data:
+            return {
+                '_type': 'url',
+                'url': data['ytBroadcastId'],
+                'ie_key': YoutubeIE.ie_key(),
+            }
+
+        # Assume HLS is available
+        media = data.get('media')
+        if media:
+            stream_url = 'https://cdn.younow.com/php/api/broadcast/videoPath/hls=1/broadcastId=%s/channelId=%s' % (
+                data.get('broadcastId'),
+                data.get('userId'),
+            )
+        else:
+            raise UnsupportedError('Unsupported stream')
+
+        webpage = self._download_webpage(url, username)
+
+        return {
+            'id': data['user']['profileUrlString'],
+            'is_live': True,
+            'title': data.get('title'),
+            'description': self._og_search_description(webpage),
+            'thumbnail': data.get('awsUrl'),
+            'tags': data.get('tags'),
+            'categories': data.get('tags'),
+            'uploader': data['user']['profileUrlString'],
```

Comment:
None of the optional fields should break the extraction if missing. Read new extractor tutorial.